### PR TITLE
4711-Improve aggregation lookup

### PR DESCRIPTION
**Pull request by pkdash**
### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. The performance improvements can be tested by creating a large number of folders (something like 50 folders) in a resource and checking the page load time for that resource.

### CODE REVIEW

**Comment by hydrocheck** [comment]
| METRIC | VALUE |
|:-----:|:-----:|

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4712](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4712)

File: hs_composite_resource/models.py [hunk 1]
```diff
@@ -57,13 +69,56 @@ def logical_files(self):
...
+        class_name_to_query_mappings = dict()
+        class_name_to_query_mappings["GenericLogicalFile"] = self.genericlogicalfile_set.all()
+        class_name_to_query_mappings["NetCDFLogicalFile"] = self.netcdflogicalfile_set.all()
+        class_name_to_query_mappings["GeoRasterLogicalFile"] = self.georasterlogicalfile_set.all()
+        class_name_to_query_mappings["GeoFeatureLogicalFile"] = self.geofeaturelogicalfile_set.all()
+        class_name_to_query_mappings["FileSetLogicalFile"] = self.filesetlogicalfile_set.all()
+        class_name_to_query_mappings["ModelProgramLogicalFile"] = self.modelprogramlogicalfile_set.all()
+        class_name_to_query_mappings["ModelInstanceLogicalFile"] = self.modelinstancelogicalfile_set.all()
+        class_name_to_query_mappings["TimeSeriesLogicalFile"] = self.timeserieslogicalfile_set.all()
+        class_name_to_query_mappings["RefTimeseriesLogicalFile"] = self.reftimeserieslogicalfile_set.all()
```
Comment (hyi) [inline hs_composite_resource/models.py:96]: Don't use ```dict()``` to initialize a dictionary but rather use `{}` due to performance reasons. See http://katrin-affolter.ch/Python/Python_dictionaries for detailed explanation. Even better, I think you can just directly define the dict as follows: 
```
class_name_to_query_mappings = {
    "GenericLogicalFile": self.genericlogicalfile_set.all(),
   ....
}
```

File: hs_composite_resource/models.py [hunk 2]
```diff
@@ -117,18 +172,30 @@ def add_file_to_aggregation(self, moved_res_file):
...
          for which the aggregation object to be retrieved
+         :param aggregations:   list of aggregations in self (resource)
         """
+        if __debug__:
+            if aggregations is not None:
+                assert(isinstance(aggregations, list))
+                for aggr in aggregations:
+                    assert(aggr.resource == self)
+                assert(len(aggregations) == len(list(self.logical_files)))
+
```
Comment (hyi) [inline hs_composite_resource/models.py:189]: Statements above for debugging can be removed

**Review by hyi — COMMENTED** [review]
@pkdash Left a couple of comments for your consideration.

File: hs_composite_resource/models.py (hunk 1 above)
Comment (pkdash) [reply hs_composite_resource/models.py]: Fixed

File: hs_composite_resource/models.py (hunk 2 above)
Comment (pkdash) [reply hs_composite_resource/models.py:195]: Fixed

**Comment by pkdash** [comment]
@hyi I have now addressed your comments.

**Review by hyi — APPROVED** [review]

**Pull request merged**